### PR TITLE
Fix InputDecorator default hint text style on M3

### LINKED ISSUE #148787: InputDecoration.hintText default text style is not compliant with M3 spec
### Steps to reproduce

- Copy the given test to packages/flutter/test/material/input_decorator_test.dart
- Run it

### Expected results

The text succeeds because the default hint style for M3 should be bodyLarge with the color set to onSurfaceVariant.
These information are available in the M3 Figma Kit.
See this Figma screenshot taken while focusing on the 'Placeholder' text (which corresponds to hint).

image

### Actual results

The test fails

### Code sample

<details open><summary>Code sample</summary>

```dart

        testWidgets('hint style is compliant with M3 spec', (WidgetTester tester) async {
          await tester.pumpWidget(
            buildInputDecorator(
              isEmpty: true,
              decoration: const InputDecoration(
                filled: true,
                hintText: hintText,
              ),
            ),
          );

          // Hint is visible because decorator is empty.
          expect(getHintOpacity(tester), 1.0);

          final ThemeData theme = Theme.of(tester.element(findDecorator()));
          final Color expectedColor = theme.colorScheme.onSurfaceVariant;
          final TextStyle expectedStyle = theme.textTheme.bodyLarge!.copyWith(color: expectedColor);
          expect(getHintStyle(tester), expectedStyle);
        });
```

</details>

### Logs

<details open><summary>Logs</summary>

```console
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: TextStyle:<TextStyle(debugLabel: ((englishLike bodyLarge 2021).merge((blackMountainView
bodyLarge).apply)).copyWith, inherit: false, color: Color(0xff49454f), family: Roboto, size: 16.0,
weight: 400, letterSpacing: 0.5, baseline: alphabetic, height: 1.5x, leadingDistribution: even,
decoration: Color(0xff1d1b20) TextDecoration.none)>
  Actual: TextStyle:<TextStyle(debugLabel: ((englishLike titleMedium 2021).merge((blackMountainView
titleMedium).apply)).merge(unknown), inherit: false, color: Color(0x99000000), family: Roboto, size:
16.0, weight: 500, letterSpacing: 0.1, baseline: alphabetic, height: 1.5x, leadingDistribution:
even, decoration: Color(0xff1d1b20) TextDecoration.none)>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Users/bruno/Nevercode/flutter/packages/flutter/test/material/input_decorator_test.dart:2840:11)
<asynchronous suspension>
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:183:15)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1017:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/bruno/Nevercode/flutter/packages/flutter/test/material/input_decorator_test.dart line 2840
The test description was:
  hint style is compliant with M3 spec
════════════════════════════════════════════════════════════════════════════════════════════════════
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel master, 3.22.0-36.0.pre.51, on macOS 14.2.1 23C71
    darwin-arm64, locale fr-FR)
```

</details>

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PRs makes the `InputDecoration.hintText` style compliant with the M3 spec.
The hint style is not clearly specified in https://m3.material.io/components/text-fields/specs, but it is in the M3 Figma kit.
('hint' terminology came from the Material1 specification, since M2 the terminology is 'Placeholder').

See this Figma screenshot taken while focusing on the 'Placeholder' text (which corresponds to hint).

image

It seems that the intention is that the 'Placeholder' colors should be the same as the 'supporting text' ones, that is why is reused 'supporting text' tokens.

## Related Issue

Fixes https://github.com/flutter/flutter/issues/148787.

## Tests

Updates several tests.

### CODE REVIEW

**Comment by bleroux** [comment]
@Renzo-Olivares This a follow-up PR to https://github.com/flutter/flutter/pull/148113 where I added the TODOs.

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -1754,8 +1754,10 @@ class InputDecorator extends StatefulWidget {
   /// The style on which to base the label, hint, counter, and error styles
   /// if the [decoration] does not provide explicit styles.
   ///
-  /// If null, [baseStyle] defaults to the `titleMedium` style from the
-  /// current [Theme], see [ThemeData.textTheme].
+  /// If null, [TextTheme.bodyLarge] will be used
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/input_decorator.dart:1757]: nit: period at the end of this sentence.

File: dev/tools/gen_defaults/lib/input_decorator_template.dart [hunk 2]
```diff
@@ -34,9 +34,9 @@ class _${blockName}DefaultsM3 extends InputDecorationTheme {
   @override
   TextStyle? get hintStyle => MaterialStateTextStyle.resolveWith((Set<MaterialState> states) {
     if (states.contains(MaterialState.disabled)) {
-      return TextStyle(color: Theme.of(context).disabledColor);
+      return TextStyle(color: ${componentColor('md.comp.filled-text-field.disabled.supporting-text')});
```
Comment (guidezpl) [inline dev/tools/gen_defaults/lib/input_decorator_template.dart:37]: I think it's the same, but prefer `md.comp.filled-text-field.disabled.label-text`

File: dev/tools/gen_defaults/lib/input_decorator_template.dart [hunk 3]
```diff
@@ -34,9 +34,9 @@ class _${blockName}DefaultsM3 extends InputDecorationTheme {
...
     }
-    return TextStyle(color: Theme.of(context).hintColor);
+    return TextStyle(color: ${componentColor('md.comp.filled-text-field.supporting-text')});
```
Comment (guidezpl) [inline dev/tools/gen_defaults/lib/input_decorator_template.dart:39]: Prefer `md.comp.filled-text-field.label-text`

**Review by guidezpl — CHANGES_REQUESTED** [review]

**Review by guidezpl — APPROVED** [review]
Disreguard previous review, LGTM :)

**Comment by bleroux** [comment]
@Renzo-Olivares The Google tests failures are probably legit. Can you please have a look?

**Comment by Renzo-Olivares** [comment]
@bleroux I just looked through the google testing failures and they all look expected. I'll try to land this tomorrow.

**Review by Renzo-Olivares — APPROVED** [review]

**Comment by Renzo-Olivares** [comment]
@bleroux I missed the golden changes when reviewing this. I took a look https://flutter-gold.skia.org/detail?grouping=name%3Dmaterial.m3_input_decorator.outline_icon_label.rtl%26source_type%3Dflutter&digest=0ebe1c484196c24b21ab16fc22738bab but am a little unsure. It looks like the floating label is being affected which I didn't expect.

**Comment by loic-sharma** [comment]
Reason for revert: This caused Skia golden changes that might be unexpected. We can reland once we confirm these changes are expected, or, fixed the changes to the floating label.

**Pull request merged**